src/docs/get-started/codelab-web.md

```diff
@@ -572,158 +678,283 @@ The animation has the following behavior:
-and another for the color animation:
+<li markdown="1">Add an AnimatedProgressIndicator.<br>
+At the bottom of the file, add this widget:
 
 <!--skip-->
 ```dart
-...
-class _SignUpFormState extends State<SignUpForm>
-   with SingleTickerProviderStateMixin {
- AnimationController animationController;  // NEW
- Animation<Color> colorAnimation;          // NEW
- bool _formCompleted = false;
-...
-```
-</li>
+class AnimatedProgressIndicator extends StatefulWidget {
+  final double value;
 
-<li markdown="1">Initialize the `AnimationController`:
-Define the AnimationController in `initState()`:
+  AnimatedProgressIndicator({
+    @required this.value,
+  });
 
-<!--skip-->
-```dart
-class _SignUpFormState extends State<SignUpForm>
-   with SingleTickerProviderStateMixin {
- AnimationController animationController;
- Animation<Color> colorAnimation;
- bool _formCompleted = false;
-
- @override
- void initState() {
-   super.initState();
-
-   animationController = AnimationController(                      // NEW
-       vsync: this, duration: const Duration(milliseconds: 1200)); // NEW
-...
+  @override
+  State<StatefulWidget> createState() {
+    return _AnimatedProgressIndicatorState();
+  }
+}
+
+class _AnimatedProgressIndicatorState extends State<AnimatedProgressIndicator>
+    with SingleTickerProviderStateMixin {
+  AnimationController _controller;
+  Animation<Color> _colorAnimation;
+  Animation<double> _curveAnimation;
+
+  void initState() {
+    super.initState();
+    _controller = AnimationController(
+        duration: Duration(milliseconds: 1200), vsync: this);
+
+    var colorTween = TweenSequence([
+      TweenSequenceItem(
+        tween: ColorTween(begin: Colors.red, end: Colors.orange),
+        weight: 1,
+      ),
+      TweenSequenceItem(
+        tween: ColorTween(begin: Colors.orange, end: Colors.yellow),
+        weight: 1,
+      ),
+      TweenSequenceItem(
+        tween: ColorTween(begin: Colors.yellow, end: Colors.green),
+        weight: 1,
+      ),
+    ]);
+
+    _colorAnimation = _controller.drive(colorTween);
+    _curveAnimation = _controller.drive(CurveTween(curve: Curves.easeIn));
+  }
+
+  void didUpdateWidget(oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    _controller.animateTo(widget.value);
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return AnimatedBuilder(
+      animation: _controller,
+      builder: (context, child) => LinearProgressIndicator(
+        value: _curveAnimation.value,
+        valueColor: _colorAnimation,
+        backgroundColor: _colorAnimation.value.withOpacity(0.4),
+      ),
+    );
+  }
+}
 ```
 </li>
 
-<li markdown="1">Create a `Tween` sequence.<br>
-Also in `initState()`, define a `Tween` sequence consisting
-of three color tweens, one for each color of the animation
-(red-to-orange, orange-to-yellow, yellow-to-green).
+Then, replace the `LinearProgressIndicator` in the Form with this new
+`AnimatedProgressIndicator`:
```

Comment:
Scroll to the end of [step 3 in the generated page](https://sz-flutter.firebaseapp.com/docs/get-started/codelab-web#step-3-add-animation-for-sign-in-progress). The code following the codeblock should be in its own codeblock.

Reply:
fixed